performance_test.py

```diff
@@ -0,0 +1,175 @@
+import subprocess
+import datetime
+import csv
+import time
+import os
+import shutil
+import resource
+from dataclasses import dataclass
+import psutil
+import errno
+
+@dataclass
+class Result:
+    returncode: int
+    stdout: str
+    stderr: str
+    duration: float
+    rusage: resource.struct_rusage
+
+LOG_FILE = "probe_log"
+RECORD_DIR = "probe_record"
+
+class ResourcePopen(subprocess.Popen):
+    def _try_wait(self, wait_flags):
+        try:
+            (pid, sts, res) = os.wait4(self.pid, wait_flags)
+        except OSError as e:
+            if e.errno != errno.ECHILD:
+                raise
+            pid = self.pid
+            sts = 0
+        else:
+            self.rusage = res
+        return (pid, sts)
+
+def resource_call(*popenargs, timeout=None, **kwargs):
+    with ResourcePopen(*popenargs, **kwargs) as p:
+        try:
+            stdout, stderr = p.communicate(timeout=timeout)
+            return p.returncode, p.rusage, stdout, stderr
+        except:
+            p.kill()
+            stdout, stderr = p.communicate()
+            raise
+
+def cleanup():
+    if os.path.exists(LOG_FILE):
+        print("Removing log file.")
+        subprocess.run("rm probe_log", shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    if os.path.exists(RECORD_DIR):
+        print("Removing record directory.")
+        shutil.rmtree(RECORD_DIR)
+    time.sleep(3)
+
+def benchmark_command(command: list[str], warmup_iterations: int, benchmark_iterations: int, transcribe_flag: bool) -> list[Result]:
+    results = []
+
+    for _ in range(warmup_iterations):
+        print(f"Running warmup command: {' '.join(command)}")
+        subprocess.run("rm probe_log", shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
It is cleaner to use `pathlib.Path("probe_log").unlink()` instead since that does not invoke a subprocess (what if `rm` does not exist on the path?).